db/python/tables/analysis.py

```diff
@@ -308,12 +398,29 @@ async def get_latest_complete_analysis_for_type(
         rows = await self.connection.fetch_all(_query, values)
         if len(rows) == 0:
             raise NotFoundError(f"Couldn't find any analysis with type {analysis_type}")
-        a = AnalysisInternal.from_db(**dict(rows[0]))
+        latest_analysis = rows[0]
+        latest_analysis_data = dict(latest_analysis)
+        analysis_outputs_by_aid = await self.get_file_outputs_by_analysis_ids(
```

Comment:
This bit of code to attach the outputs to the analysis is repeated a few times in this file, would be good to break it out into a function to reduce the repetition.

Reply:
I will address this in a separate PR, as I noticed a lot of code in this file was reusable so it might be worth addressing it all at once :) Super useful suggestion, thanks!